### PR TITLE
Adding support for podman play volumes

### PR DESCRIPTION
**Pull request by aidan-plenert-macdonald**
To resolve https://github.com/containers/podman/issues/5788

### CODE REVIEW

File: pkg/domain/infra/abi/play_test.go [hunk 1]
```diff
@@ -252,3 +258,86 @@ func TestEnvVarValue(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerEngine_PlayKube(t *testing.T) {
```
Comment (aidan-plenert-macdonald) [inline pkg/domain/infra/abi/play_test.go:262]: This test doesn't work. I need to mock out the runtime somehow.

**Comment by rhatdan** [comment]
Thanks @aidan-plenert-macdonald 
You need to sign your PRs
git commit -a -s --amend
git push --force

**Comment by mheon** [comment]
We generally avoid unit tests due to the complexities of mocking the increasingly-large Runtime; instead, we use extensive E2E tests of the functionality in question using a full Podman executable.

**Comment by aidan-plenert-macdonald** [comment]
Okay. That's fair. This might answer my question in another issue I opened.
I was offering to rework some of the code to make it easier to unit test.
If you don't want that, I'll just do e2e tests. They are just slower, so I
typically do unit tests first.

**Comment by rhatdan** [comment]
@aidan-plenert-macdonald We would love to have more unit tests, but some are just too difficult to setup, so we fall back to E2E.

**Comment by xordspar0** [comment]
This is a great start to #5788 and I look forward to being able to make volumes with k8s config. This is only part of what's needed for that issue though. This PR allows creating volumes, but it doesn't allow mounting them into a container. We should also add support for volumes to `generate`.

I think it makes sense to keep the scope of this issue as it is, but also keep track of these other features that still need to be added.

@aidan-plenert-macdonald, the smoke tests are failing due to a couple git commit issues. Could you please squash your commits into a single commit, make sure the squashed commit still has your DCO signoff, and make sure the length of the commit title is 90 characters or less? For the commit title, I would suggest "Add support for podman play volumes". See https://chris.beams.io/posts/git-commit/

Thanks for working on this and I look forward to seeing this merged!

**Comment by aidan-plenert-macdonald** [comment]
Should have marked a draft. I'm still trying to get the e2e tests to pass locally, then I'll do the `generate` support and make sure it actually works. Thanks for checking!

**Comment by vrothberg** [comment]
@aidan-plenert-macdonald, are you still working on it?

**Comment by EduardoVega** [comment]
I am currently working on #9129 and both issues change the same files. So it might be nice to have this one first. If there is no activity on this one, I can continue with the work / changes that are pending.

**Comment by rhatdan** [comment]
@EduardoVega Take this over if you can.

**Comment by EduardoVega** [comment]
I'll do. Thanks

**Comment by mheon** [comment]
@EduardoVega Good to close, given #9935 has merged?

**Comment by EduardoVega** [comment]
@mheon correct.

**Pull request closed without merging**